pkg/configure/oneagent/pmc/ruxit/from.go

```diff
@@ -0,0 +1,95 @@
+package ruxit
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// example match: [general]
+var sectionRegexp = regexp.MustCompile(`\[(.*)\]`)
+
+func FromMap(procMap ProcMap) ProcConf {
+	var result ProcConf
+
+	for section, props := range procMap {
+		for key, value := range props {
+			result.Properties = append(result.Properties, Property{
+				Section: section,
+				Key:     key,
+				Value:   value,
+			})
+		}
+	}
+
+	return result
+}
+
+func FromJson(reader io.Reader) (ProcConf, error) {
+	var result ProcConf
+
+	raw, err := io.ReadAll(reader)
+	if err != nil {
+		return result, errors.WithStack(err)
+
+	}
+
+	err = json.Unmarshal(raw, &result)
+	if err != nil {
+		return result, errors.WithStack(err)
+
+	}
+
+	return result, nil
+}
+
+// FromConf creates the ProcConf struct from an valid ruxitagentproc.conf config file.
+func FromConf(reader io.Reader) (ProcConf, error) {
+	var result []Property
+
+	scanner := bufio.NewScanner(reader)
+	currentSection := ""
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		header := confSectionHeader(line)
+
+		switch {
+		case header != "":
+			currentSection = strings.Trim(header, "\t\n\v\f\r ")
```

Comment:
Should we safe `"\t\n\v\f\r "` into a constant?

Reply:
done